docs/guide/core-concepts.md

```diff
@@ -0,0 +1,427 @@
+Naively, this means somewhere around 63 times less data than if the most
+compact form possible had been chosen (the use of `BitCells` would
+be maximally efficient for representing the bivalent nature of boolean
+values). See the chart below for a sense of the relative sizes of these
+cell types.  
+
+The performance impacts of cell type selection matter in both a local
+and a distributed (spark) context. Locally, the memory footprint will mean
+that as larger cell types are used, smaller amounts of data can be held in
+memory and worked on at a given time and that more CPU cache misses are to be
+expected. This latter point - that CPU cache misses will increase - means that
+more time spent shuffling data from the memory to the processor (which
+is often a performance bottleneck). When running programs that
+leverage spark for compute distribution, larger data types mean more
+data to serialize and more data send over the (very slow, relatively
+speaking) network.  
+
+In the chart below, `DataType`s are listed in the leftmost column and
+important characteristics for deciding between them can be found to the
+right. As you can see, the difference in size can be quite stark depending on
+the cell type that a tile is backed by. That extra space is the price
+paid for representing a larger range of values. Note that bit cells
+lack the sufficient representational resources to have a `NoData` value.  
+
+|             | Bits / Cell | 512x512 Raster (mb) |     Range (inclusive)     | GeoTrellis NoData Value |
+|-------------|:-----------:|---------------------|:-------------------------:|-------------------------|
+| BitCells    | 1           | 0.032768            | [0, 1]                    |                     N/A |
+| ByteCells   | 8           | 0.262144            | [-128, 128]               |                    -128 |
+| UbyteCells  | 8           | 0.262144            | [0, 255]                  |                       0 |
+| ShortCells  | 16          | 0.524288            | [-32768, 32767]           |                  -32768 |
+| UShortCells | 16          | 0.524288            | [0, 65535]                |                       0 |
+| IntCells    | 32          | 1.048576            | [-2147483648, 2147483647] |             -2147483648 |
+| FloatCells  | 32          | 1.048576            | [-3.40E38, 3.40E38]       |               Float.NaN |
+| DoubleCells | 64          | 2.097152            | [-1.79E308, 1.79E308]     |              Double.NaN |
+
+One final point is worth making in the context of `CellType`
+performance: the `Constant` types are able to depend upon macros which
+inline comparisons and conversions. This minor difference can certainly
+be felt while iterating through millions and millions of cells. If possible, Constant
+`NoData` values are to be preferred. For convenience' sake, we've
+attempted to make the GeoTrellis-blessed `NoData` values as unobtrusive
+as possible a priori.  
```

Comment:
Consider discussing (or linking to a discussion elsewhere) of how RDD's are different from tiles:
- Cell type is stored internally to a tile, but externally to an RDD (in its metadata)
- Must use `withContext` or `ContextRDD()` to preserve metadata in RDD operations